index.d.ts

```diff
@@ -0,0 +1,149 @@
+import { EventEmitter } from "events";
+import http from 'http';
+
+/**
+ * arguments of constructor which in class extend DataStore
+ */
+declare interface DataStoreOptions{
+    path: string;
+    namingFunction?: (req : http.IncomingMessage) => string;
+    relativeLocation?: string;
+}
+
+declare interface FileStoreOptions extends DataStoreOptions{
+    directory?: string;
+}
+
+declare interface GCStoreOptions extends DataStoreOptions{
+    bucket: string;
+    projectId: string;
+    keyFilename: string;
+}
+
+declare interface S3StoreOptions extends DataStoreOptions{
+    accessKeyId: string;
+    secretAccessKey: string;
+    bucket: string;
+    tmpDirPrefix: string;
+    partSize: number;
+}
+
+declare class File{
+    id: string;
+    upload_length: any;
+    upload_defer_length: any;
+    upload_metadata: any;
+    constructor(file_id: string, upload_length: any, upload_defer_length: any, upload_metadata: any);
+}
+
+/**
+ * Based store for all DataStore classes.
+ */
+export declare class DataStore extends EventEmitter{
+    constructor(options : DataStoreOptions);
+    get extensions() : any;
+    set extensions(extensions_array : any);
+    create(req: http.IncomingMessage): Promise<any>;
+    write(req: http.IncomingMessage, file_id?: string, offset?: number): Promise<any>;
+    getOffset(id: string): Promise<any>;
+}
+
+/**
+ * file store in local storage
+ */
+export declare class FileStore extends DataStore{
+    constructor(options : FileStoreOptions);
+    _checkOrCreateDirectory(): any;
```

Comment:
Why are the private methods (prefixed with an underscore) included in here? Is this necessary?

Reply:
I remove all method which starts with _